test/replication-luatest/gh_9235_assertion_in_box_wait_limbo_acked_test.lua

```diff
@@ -0,0 +1,112 @@
+    -- instances are in 'connect' state instead of 'follow'.
+    t.helpers.retrying({timeout = wait_timeout}, function()
+        server1:wait_for_vclock_of(server2)
+        server2:wait_for_vclock_of(server1)
+        server1:assert_follows_upstream(server2:get_instance_id())
+        server2:assert_follows_upstream(server1:get_instance_id())
+    end)
+end
+
+local function server_wait_wal_is_blocked(server)
+    server:exec(function(wait_timeout)
+        t.helpers.retrying({timeout = wait_timeout}, function()
+            t.assert(box.error.injection.get('ERRINJ_WAL_DELAY'))
+        end)
+    end, {wait_timeout})
+end
+
+local function server_wait_synchro_queue_len_is_equal(server, expected)
+    server:exec(function(expected, wait_timeout)
+        t.helpers.retrying({timeout = wait_timeout}, function(expected)
+            t.assert_equals(box.info.synchro.queue.len, expected)
+        end, expected)
+    end, {expected, wait_timeout})
+end
+
+g.before_each(function(cg)
+    t.tarantool.skip_if_not_debug()
+
+    cg.cluster = cluster:new({})
+    cg.master = cg.cluster:build_and_add_server({
+        alias = 'master',
+        box_cfg = {
+            replication = {
+                server.build_listen_uri('master', cg.cluster.id),
+                server.build_listen_uri('replica', cg.cluster.id),
+            },
+            election_mode = 'candidate',
+            replication_synchro_quorum = 2,
+            replication_synchro_timeout = 100000,
+        }
+    })
+    cg.replica = cg.cluster:build_and_add_server({
+        alias = 'replica',
+        box_cfg = {
+            replication = {
+                server.build_listen_uri('replica', cg.cluster.id),
+                server.build_listen_uri('master', cg.cluster.id),
+            },
+            election_mode = 'off',
+            replication_synchro_quorum = 2,
+            replication_synchro_timeout = 100000,
+        }
+    })
+    cg.cluster:start()
+    cg.master:wait_until_election_leader_found()
+    cg.replica:wait_until_election_leader_found()
+    cg.master:exec(function()
+        box.schema.space.create('test', {is_sync = true})
+        box.space.test:create_index('pk')
+    end)
+    wait_pair_sync(cg.replica, cg.master)
+end)
+
+g.after_each(function(cg)
+    cg.cluster:drop()
+end)
+
+g.test_assert_last_entry_lsn_is_positive = function(cg)
+    local f = cg.replica:exec(function()
+       box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+       local f = require('fiber').create(function() box.ctl.promote() end)
+       box.cfg{wal_queue_max_size=1}
+       f:set_joinable(true)
+       return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.master:exec(function()
+        require('fiber').create(function()
+            box.begin()
+            require('fiber').create(function() box.space.test:insert{1} end)
+            require('fiber').create(function() box.space.test:insert{2} end)
+            box.commit()
```

Comment:
Why are two statements needed here? It seems one `insert` would do the job, no?

Reply:
I looked closely and realized that there were actually some things that could be removed here. You can remove the outer fiber, as well as `box.begin()` and `box.commit()`. They are optional. However, there must be more than one insert statement. The first one starts writing, but stops because of `ERRINJ_WAL_DELAY`, and the second one goes into the waiters queue.